core/src/main/java/org/elasticsearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -87,10 +88,10 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
         builder.endObject();
     }
 
-    public static ConstantScoreQueryBuilder fromXContent(QueryParseContext parseContext) throws IOException {
+    public static Optional<ConstantScoreQueryBuilder> fromXContent(QueryParseContext parseContext) throws IOException {
         XContentParser parser = parseContext.parser();
 
-        QueryBuilder query = null;
+        Optional<QueryBuilder> query = Optional.empty();
```

Comment:
I think you can leave null here. this init empty value will never be used as we throw exception when queryFound is false. Otherwise there are two places that can cause empty and that may be confusing.